storage.go

```diff
@@ -1,35 +1,46 @@
-package dynamodbstorage
+package skydbstorage
 
 import (
 	"context"
 	"encoding/base64"
+	"encoding/json"
 	"errors"
 	"fmt"
-	"log"
+	"strings"
 	"time"
 
-	"github.com/aws/aws-sdk-go/aws"
-	"github.com/aws/aws-sdk-go/aws/session"
-	"github.com/aws/aws-sdk-go/service/dynamodb"
-	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
+	"github.com/skynetlabs/certmagic-storage-skydb/skydb"
 
-	caddy "github.com/caddyserver/caddy/v2"
+	"github.com/caddyserver/caddy/v2"
 	"github.com/caddyserver/certmagic"
+	"go.sia.tech/siad/crypto"
 )
 
 const (
-	contentsAttribute    = "Contents"
-	primaryKeyAttribute  = "PrimaryKey"
-	lastUpdatedAttribute = "LastUpdated"
-	lockTimeoutMinutes   = caddy.Duration(5 * time.Minute)
-	lockPollingInterval  = caddy.Duration(5 * time.Second)
+	lockTimeoutMinutes  = caddy.Duration(5 * time.Minute)
+	lockPollingInterval = caddy.Duration(5 * time.Second)
+
+	// keyListDataKeyString points to a global list of known keys. We need this
+	// list for the List functionality. The list will be updated each time we
+	// Store or Delete a key.
+	keyListDataKeyString = "FvyIax1rlzkpOKsGHGYO4qi/bNgXnjvWFpNXXq13hRc="
+)
+
+var (
+	// The registry doesn't support DELETE as an operation. The next best thing
+	// is to write an empty record to it. This is the empty record that we're
+	// going to be writing.
+	emptyRegistryEntry = [34]byte{}
```

Comment:
You can use `skymodules.Skylink{}`

https://gitlab.com/SkynetLabs/skyd/-/merge_requests/102/diffs#64591b5162b9bedb750031171cf39d8d8653cd4e_661_663

Reply:
or expose `rawSkylinkSize` in skyd (f/u)